code/game/objects/items/devices/mirage.dm

```diff
@@ -0,0 +1,55 @@
+/obj/item/explosive/grenade/mirage
+	name = "mirage grenade"
+	desc = "A special device that, when activated, produces a pair of holographic copies of the user."
+	icon_state = "delivery"
+	item_state = "delivery"
+	///the parent to be copied
+	var/mob/living/current_user
```

Comment:
You could get away with only copying the appearance and faction, so you don't have to keep a reference to the mob which has to be cleaned, or worry about it qdeleting before you are done with it.

Reply:
It updates the appearance now so this is needed now

Reply:
Then register on its qdeletion in case it's deleted before the mirage, and clean the reference on `Destroy()`